### ISSUE TITLE
[bug] addVictoryInterpolator > d3Interpolate.values undefined

### ISSUE THREAD

**Issue by jonathanewerner** [FormidableLabs/victory#250]
Hi, 
I'm not sure if this is an bug or sth on my end. 
With victory 0.6.1, this code in victory-animation/utils is erroneous:

```
var addVictoryInterpolator = exports.addVictoryInterpolator = function addVictoryInterpolator() {
  if (!interpolatorAdded) {
    _d3Interpolate2.default.values.push(victoryInterpolator);
    interpolatorAdded = true;
  }
```

Specifically `_d3Interpolate2.default.values`: values is [not being exported by `d3-interpolate`](https://github.com/d3/d3-interpolate/blob/master/index.js), which leads to 

```
util.js?ee21:183 Uncaught TypeError: Cannot read property 'push' of undefined 
  @ addVictoryInterpolator @ util.js?ee21:183(anonymous function) 
  @ victory-animation.js?385f:35(anonymous function) @ app.js:9112__webpack_require__ @ app.js:556fn 
```

Any pointers?

**Comment by boygirl** [comment]
This looks to me like a mismatch between versions of d3-interpolate. The concept of an array of values or interpolators was removed in recent versions, and the d3 modules are all on pre 1.0 semver (i.e. minor version bumps for breaking changes). That plus npm dependency flattening could put you in an annoying situation.

**Comment by jonathanewerner** [comment]
Does that mean that this is the fault of a version not being specified exactly in some dependent package of victory?

**Comment by boygirl** [comment]
I will triple check that, but I believe we are specifying an exact version of d3-interpolate everywhere we are using it. I'll help you figure out what is causing this issue so it doesn't cause problems for others.

**Comment by vyorkin** [comment]
I've got the same problem. I'm using:
```
"victory": "0.7.0",
"d3-scale": "0.7.0",
```

**Comment by vyorkin** [comment]
I've tried to remove `d3-scale` and now I've got this error:
```
Module not found: Error: Cannot resolve module 'd3-interpolate' in /my-long-path-here/node_modules/victory-core/lib/victory-animation
resolve module d3-interpolate in /my-long-path-here/node_modules/victory-core/lib/victory-animation
 ```

**Comment by boygirl** [comment]
Can you try removing d3-scale and victory from node_modules and removing d3-scale from package.json and running npm install again?

**Comment by vyorkin** [comment]
I've already tried that today morning – no luck, have no idea what can be wrong. Here is what I did:
```
rm -rf node_modules/d3*
rm -rf node_modules/victory*
```
than I removed `d3-scale` from the `package.json` and ran `npm i` and after that the same error remains

**Comment by boygirl** [comment]
I have to admit to being very confused by this issue. I'll try to reproduce it locally.

**Comment by hitchcott** [comment]
I'm also getting stung by this using `gatsby`, but it seemed to work fine before using `webpack 2`. Could it have something to do with that?

```
└─┬ victory@0.8.0
  ├─┬ victory-chart@9.0.0
  │ ├─┬ d3-scale@0.6.4
  │ │ ├── d3-array@0.7.1
  │ │ ├── d3-collection@0.1.2
  │ │ ├── d3-color@0.4.2
  │ │ ├── d3-format@0.5.1
  │ │ ├── d3-interpolate@0.7.0
  │ │ ├── d3-time@0.2.5
  │ │ └── d3-time-format@0.3.2
  │ └─┬ d3-shape@0.6.1
  │   └── d3-path@0.1.5
  ├─┬ victory-core@3.0.0
  │ ├── d3-ease@0.3.1
  │ ├─┬ d3-interpolate@0.2.0
  │ │ └── d3-color@0.3.4
  │ └── d3-timer@0.0.6
  └── victory-pie@4.0.0
```

```
TypeError: Cannot read property 'push' of undefined
    at addVictoryInterpolator (webpack:///../~/victory-core/lib/victory-animation/util.js?:183:35)
    at eval (webpack:///../~/victory-core/lib/victory-animation/victory-animation.js?:35:34)
    at Object.<anonymous> (http://localhost:8000/bundle.js?t=1464892614382:2416:2)
    at __webpack_require__ (http://localhost:8000/bundle.js?t=1464892614382:556:30)
    at fn (http://localhost:8000/bundle.js?t=1464892614382:87:20)
    at eval (webpack:///../~/victory-core/lib/index.js?:62:25)
    at Object.<anonymous> (http://localhost:8000/bundle.js?t=1464892614382:640:2)
    at __webpack_require__ (http://localhost:8000/bundle.js?t=1464892614382:556:30)
    at fn (http://localhost:8000/bundle.js?t=1464892614382:87:20)
    at eval (webpack:///../~/victory/lib/index.js?:8:20)
```

**Comment by ryan-roemer** [comment]
@hitchcott -- Do you have a publicly available repository / branch that we can reproduce this from? Thanks!

**Comment by ryan-roemer** [comment]
Also, if you switch to using the built victory (https://npmcdn.com/victory@0.8.0/dist/victory.js) with something like:

```js
var victory = require("victory/dist/victory");
```

in code, OR in your webpack config add something like:

```js
resolve: {
  alias: {
    victory: require.resolve("victory/dist/victory")
  }
}
```

Do you still get the error?

**Comment by tutturen** [comment]
I got the same bug today right after i added a resolver to my webpack config.
Adding the alias @ryan-roemer mentioned fixed it.

**Comment by okonet** [comment]
I can confirm this one. Importing from `victory/dist/victory` solves it.

**Comment by ryan-roemer** [comment]
Reviewing the comments here it looks like webpack configurations somehow _aren't_ picking up the right version of `d3-interpolate` despite being correctly specified in the upstream components, e.g. https://github.com/FormidableLabs/victory-animation/blob/master/package.json#L26

Could the folks hitting the issue post package.json deps / devDeps here? Or, even better, a repo / branch that hits the issue for us to maybe have a look at?

I don't think it's a victory problem, but worth looking at since only using the `dist` version of victory certainly isn't a desired prod-facing solution.

Random guess here, but the huge, huge npm dependency issue I see cropping up everywhere these days is: https://github.com/npm/npm/issues/10727 wherein a single `npm install` doesn't install **all** the dependency tree for some dedupes. Then you end up with the wrong version of a lib because node traverses up the tree. For anyone with a reproducible error, could you report on the following experiment: (1) `npm install` normally, (2) see the problem, (3) without uninstalling `node_modules` or anything, run a **second** `npm install` right over the top of the previous one, and (4) report if the issue persists?

**Comment by lropero** [comment]
I just learned something from my Webpack configuration that might shed some light on the "Cannot read property 'push' of undefined" fatal error message that forces some of us to require the dist version of victory (I just fixed it, keep reading).

Some time ago I noticed this unrelated-to-this-issue warning whenever I started webpack-dev-server:

```
WARNING in (webpack)/buildin/module.js
There is another module with an equal name when case is ignored.
This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.
Rename module if multiple modules are expected or use equal casing if one module is expected.
```

This warning is widely discussed [here](https://github.com/JedWatson/classnames/issues/82) and [here](https://github.com/webpack/webpack/issues/382) so I won't enter into details about it now, but it turns out that if I add the following to webpack.config.js the warning goes away:

```
resolve: {
	modulesDirectories: [
		path.join(__dirname, 'node_modules')
	]
}
```

Having this line produced the "Cannot read property 'push' of undefined" error message that required me to add the victory alias to its dist release (in webpack's resolve object as well).

So I removed all these resolves and went back to seeing the aforementioned warning, but now victory works well. This also fixed [this issue](https://github.com/FormidableLabs/victory/issues/297) I was struggling with.

Hope it helps.
